BondageClub/Scripts/Common.js

```diff
@@ -9,7 +9,7 @@ var CurrentModule;
 var CurrentScreen;
 /** @type {ScreenFunctions} */
 var CurrentScreenFunctions;
-/** @type {Character|null} */
+/** @type {Character|NPCCharacter|null} */
```

Comment:
`NPCCharacter` won't do anything here, as it already extends `Character`.
This means, that because `null` is ignored (we are not in strict mode), the largest intersection of all types here is still `Character`.
```suggestion
/** @type {Character|null} */
```

Reply:
Huh, but you can assign a more specific `NPCCharacter` instance in here, and then all of its subclass-specific property accesses will warn because the "compiler" think its a `Character` and doesn't see them?